### DISCUSSION TITLE
Is there an OpenGL wrapper for V?

### DISCUSSION THREAD

**Discussion by dragsbruh** [vlang/v#19187] [Questions and Answers]
I am a game developer who wants to make my own graphics library using opengl. But I cannot find any opengl wrapper for V. Did I miss something or are there actually no opengl wrappers for V? I saw you made gg in opengl, so it should be there right?

**Comment by medvednikov** [comment]
There used to be, I'll bring it back. For now you can use gg/sokol/sdl2

**Reply by dragsbruh** [reply]
Did you bring it back yet? I'm excited

**Comment by renlite** [comment]
Is OpenGL wrapper back?

**Reply by dragsbruh** [reply]
not yet, if medvednikov did not forget about it

**Reply by JalonSolov** [reply]
You can always try the previous code.  It just hasn't been updated in a while, and may not even compile with current V.

But at least you won't be starting from scratch...

https://github.com/vlang/glfw

**Comment by noxomix** [comment]
I also created a wrapper few months ago, but its just tested with Linux. Also keep in mind, the wrapper is auto generated which means Ive written a script which has then written all the wrapper code lines from the official Khronos OpenGL XML registry. So I didnt checked if all the function bindings work correctly, there might be bugs which can trigger `Segmentation Fault` errors , but until now everything worked for me (2D rendering). It supports every openGL function until 4.6 but by default its shipped with OpenGL 2.1 i guess. So if you want a newer version of it you just replace the `gl.h` with a GLAD-generated header-only file and rename it as gl.h. (https://gen.glad.sh/ and select your opengl version and check the header-only option) the bindings will recognize which version so you dont have to do anything else then replace the gl.h and vGL will do the rest. 

https://github.com/noxomix/vGL

**Comment by renlite** [comment]
Roadmap [Version 0.5]: "Direct C/C++ interop without generating wrappers" should help to use the C APIs for GL and GLFW direct.
https://github.com/vlang/v/blob/master/ROADMAP.md

**Comment by hholst80** [comment]
If you would humor the idea of starting from scratch, consider porting Raylib to V ;-)  The same mentioned C interop features would ofc help significantly.

https://www.raylib.com/

**Reply by JalonSolov** [reply]
There are already several raylib wrappers for V.  See https://github.com/vlang/awesome-v#game-development

https://github.com/irishgreencitrus/raylib.v is usually rated as the best.
